packages/core/src/builder.class.ts

```diff
@@ -2119,7 +2121,9 @@ export class Builder {
     if (res) {
       this.response = res;
     }
-    this.canTrack = canTrack;
+    if (!this.hasOverriddenCanTrack) {
+      this.canTrack = canTrack;
```

Comment:
Ah, good catch! `this.defaultCanTrack` is overriding `builder.canTrack()` calls made before `builder.init()`.

Could you add a comment explaining that here? Something like:

> If `builder.canTrack` was set before the `builder.init()` call, we want to preserve the user's choice and not override it with a default.